pkg/kubelet/kubelet.go

```diff
@@ -1152,120 +1152,183 @@ func (kl *Kubelet) pullImageAndRunContainer(pod *api.BoundPod, container *api.Co
 	return containerID, nil
 }
 
-func (kl *Kubelet) syncPod(pod *api.BoundPod, containersInPod dockertools.DockerContainers) error {
+type podContainerChangesSpec struct {
+	StartInfraContainer bool
+	InfraContainerId    dockertools.DockerID
+	ContainersToStart   map[int]empty
```

Comment:
That was my original idea. I run into the problem that I don't really understand pointer semantics in Go. What I found out is that when you iterate over a slice you get a copy of an element, not the reference to it, and it's being reused in each iteration (at least that's the kind of behaviour I was seeing). I may try getting the pointer directly by &vector[index], but then I'm kind of scared what will happen if runtime will decide that it'll resize the slice...

In other words: in C++ it'd have been a pointer from the start, in Go I'm a bit afraid